pkgs/development/libraries/numcpp/default.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  boost181,
+  python3,
+  gtest,
+  eigen,
+  opencv,
+}:
+let
+  boost = boost181.override { enableStatic = true; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "numcpp";
+  version = "2.11.0";
+
+  src = fetchFromGitHub {
+    owner = "dpilger26";
+    repo = "NumCpp";
+    rev = "Version_${finalAttrs.version}";
+    hash = "sha256-IAku1bcaMkawZxpQbvxcS6VX07ogw4UGo1DX2Wa8xwU=";
+  };
+
+  nativeBuildInputs = [cmake gtest python3 eigen opencv];
+
+  cmakeFlags = ["-DBUILD_ALL=ON"];
```

```suggestion
  cmakeFlags = lib.optionals finalAttrs.doCheck [
    "-DBUILD_TESTS=ON"
    "-DBUILD_MULTIPLE_TEST=ON"
  ];
```

I'm not sure there's a lot of value in building the examples, especially if they drag in huge build-time dependencies like opencv.

---

okay now just doing tests

---

Looks like this should actually be `finalAttrs.finalPackage.doCheck` so that it automatically disables itself when cross-compiling.